tests

```diff
@@ -294,6 +306,141 @@ EOF
 	./safe export secret/import >t/home/got 2>t/home/errors
 	yamlok
 
+	testing ${version} x509 commands
+	./safe x509 issue                                                ; exitok $? 1 # need a path
+	./safe x509 issue --signed-by secret/enoent                      ; exitok $? 1 # still need a name...
+	./safe x509 issue --signed-by secret/enoent secret/cert          ; exitok $? 1 # ca must exist
+
+	./safe x509 issue secret/x509/self-signed --name www.example.com ; exitok $? 0
+	./safe x509 validate secret/x509/self-signed                     ; exitok $? 0
+	./safe exists secret/x509/self-signed:certificate                ; exitok $? 0
+	./safe exists secret/x509/self-signed:key                        ; exitok $? 0
+	./safe exists secret/x509/self-signed:combined                   ; exitok $? 0
+	./safe exists secret/x509/self-signed:serial                     ; exitok $? 1  # not a CA
+	./safe exists secret/x509/self-signed:crl                        ; exitok $? 1  # not a CA
+	./safe x509 check secret/x509/self-signed                        ; exitok $? 0
+	./safe x509 check secret/x509/self-signed --for www.example.com  ; exitok $? 0
+	./safe x509 check secret/x509/self-signed --for mail.example.com ; exitok $? 1
+	./safe x509 check secret/x509/self-signed --for test@example.com ; exitok $? 1
+	./safe x509 check secret/x509/self-signed --for 10.40.5.6        ; exitok $? 1
+
+	./safe x509 issue secret/fail-without-name                       ; exitok $? 1
+	./safe exists secret/fail-without-name:certificate               ; exitok $? 1
+	./safe exists secret/fail-without-name:key                       ; exitok $? 1
+	./safe exists secret/fail-without-name:combined                  ; exitok $? 1
+	./safe exists secret/fail-without-name:serial                    ; exitok $? 1
+	./safe exists secret/fail-without-name:crl                       ; exitok $? 1
```

Comment:
I'm more asking what the expected behaviour is --  an over-write test could be written as an addendum to an existing test to either confirm that after a test has successfully created a cert or ca, it  a) continues to exist after a failure to create a new one at the same path, or b) deletes the existing cert/ca because it failed to create one.  I suggest the correct behaviour is b, because it is arguably safer security-wise.  There is a further black-box test to confirm that if you create a cert in a path that once held a ca, there is no longer a serial or crl element stored on that path. (I say blackbox because I'm sure that if you know the code, it would be self-evident that it won't contain those elements, but its valuable because implementations can change)